lib/spack/spack/schema/config.py

```diff
@@ -70,6 +70,7 @@
                 'dirty': {'type': 'boolean'},
                 'build_jobs': {'type': 'integer', 'minimum': 1},
                 'ccache': {'type': 'boolean'},
+                'file_lock_timeout': {'type': 'integer', 'minimum': 1},
```

Comment:
you should probably add this to the default config file (with the default 120 seconds or commented out), as this is the first point where I'd look for settings

Reply:
I do not have this locking problem on our smaller CentOS Spack instance.  Could locking timeout be affected by the size of Spack database?

Reply:
@scheibelp @tgamblin ?

Reply:
>  Could locking timeout be affected by the size of Spack database?

I consider that unlikely. Under the hood there are two types of locks:

* Whole-database locks: managed as a lock on an individual file in `.spack-db/lock` (or thereabouts). This should take a constant amount of time regardless of the DB size
* Per-package locks: locks on the dag hash of a spec; this is managed as a byte-range lock and the number of such locks acquired is related to the size of the current DAG than to the size of the database as a whole. It could be a problem when installing large DAGs come to think of it but even the largest ones I've heard of (120 nodes) still don't seem like a lot of locking activity

> I do not have this locking problem on our smaller CentOS 7 Spack instance

Is your other Spack instance also CentOS 7? Or is it another version? Are both systems using the same filesystem type? IIRC you mentioned that both are lightly-used (i.e. not stressed from constant activity).

Reply:
@scheibelp our spec where locking fails contains O(290) nodes. At least
```
spack spec defaults | wc -l
299
```
But again it is only intermittent